tianshou/data/batch.py

```diff
@@ -24,28 +23,45 @@ def _is_batch_set(data: Any) -> bool:
     return False
 
 
-def _create_value(inst: Any, size: int) -> Union[
+def _create_value(inst: Any, size: int, stack=True) -> Union[
         'Batch', np.ndarray, torch.Tensor]:
+    """
+    :param bool stack: whether to stack or to concatenate. E.g. if inst has
+        shape of (3, 5), size = 10, stack=True returns an np.ndarry with shape
+        of (10, 3, 5), otherwise (10, 5)
+    """
+    has_shape = isinstance(inst, (np.ndarray, torch.Tensor))
+    is_scalar = \
+        isinstance(inst, Number) or \
+        issubclass(inst.__class__, np.generic) or \
+        (has_shape and not inst.shape)
+    if not stack and is_scalar:
+        # here we do not consider scalar types, following the
+        # behavior of numpy which does not support concatenation
+        # of zero-dimensional arrays (scalars)
+        raise TypeError(f"cannot cat {inst} with which is scalar")
```

Comment:
Perhaps it would be better to raise this exception at `cat` and `stack` level ? I know it is kind of code duplication, but in the particular case of raising an exception it seems appropriate to simplify the trace, since `_create_value` is not an "official" method of Tianshou and only used to factorize the code. Moreover, only `cat` can raise this exception. But as you want.

Reply:
Yeah, the trace will start with ``_create_value ``, but followed by ``cat``. This should be okay. I have seen many exceptions (like numpy) starting with an internal method. Users should be able to follow the trace to find the root cause.